File: src/main/java/com/leijendary/spring/iamtemplate/config/PropertiesConfiguration.java
package com.leijendary.spring.iamtemplate.config;

import com.leijendary.spring.iamtemplate.config.properties.*;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableConfigurationProperties({
        AspectProperties.class,
        AuthProperties.class,
        CorsProperties.class,
        InfoProperties.class,
        RoleProperties.class,
        VerificationProperties.class })
public class PropertiesConfiguration {
}

File: src/main/kotlin/com/leijendary/spring/template/iam/core/config/PropertiesConfiguration.kt
package com.leijendary.spring.template.iam.core.config

import com.leijendary.spring.template.iam.core.config.properties.*
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Configuration

@Configuration
@EnableConfigurationProperties(
    AuthProperties::class,
    AwsCloudFrontProperties::class,
    AwsS3Properties::class,
    DataSourcePrimaryProperties::class,
    DataSourceReadonlyProperties::class,
    InfoProperties::class,
    KafkaTopicProperties::class,
    VerificationProperties::class,
)
class PropertiesConfiguration 
